Update technical debt

mcornaton opened the pull request.

mcornaton committed:
Update deprecated calls

james-toussaint commented on src/main/java/com/iexec/common/chain/IexecHubAbstractService.java:
```diff
@@ -913,8 +888,8 @@ public String getTaskResults(String chainTaskId, Integer chainId) {
             if (!StringUtils.isEmpty(chainDealId)
-                    && BytesUtils.isBytes32(BytesUtils.stringToBytes(chainDealId))
-                    && !BytesUtils.EMPTY_HEXASTRING_64.equals(chainDealId)){
+                    && BytesUtils.isBytes32(chainDealId)
```
Thanks for the update which is right, how about `isNonZeroedBytes32` ?

mcornaton replied:
Sure, much cleaner. Thanks!

james-toussaint commented on src/main/java/com/iexec/common/chain/eip712/EIP712Utils.java:
```diff
@@ -72,7 +72,7 @@ static String encodeBigInteger(BigInteger bigInteger) {
 
     static String encodeByteArray(byte[] byteArray) {
-        if (!BytesUtils.isBytes32(byteArray)) {
+        if (!BytesUtils.isBytes32(BytesUtils.bytesToString(byteArray))) {
```
`BytesUtils.bytesToString(byteArray)` calls `Numeric.toHexString(bytes)`, so it looks we are calling twice the method until the `return`.

I might say checking `bytes != null && bytes.length == BYTES_32_SIZE` in the `if ` would be more appropriate

mcornaton replied:
Sure, thanks!

mcornaton committed:
Update `EIP712Challenge`

jbern0rd commented on src/main/java/com/iexec/common/chain/eip712/EIP712Entity.java:
```diff
@@ -43,7 +45,7 @@ protected EIP712Entity(EIP712Domain domain, M message) {
     @Override
     public HashMap<String, List<TypeParam>> getTypes() {
-        return types;
+        return new HashMap<>(types);
```
```suggestion
        return types.clone();
```
Would it be more readable this way ???

mcornaton replied:
I prefer to create a new map by hand instead of using the `clone` method. I find it a bit cleaner.

mcornaton committed:
Remove useless `public` modifier in tests

jbern0rd commented on src/main/java/com/iexec/common/chain/eip712/EIP712Entity.java:
```diff
@@ -76,4 +78,13 @@ public String signMessage(ECKeyPair ecKeyPair) {
+
+    @JsonIgnore
+    public List<TypeParam> getMessageTypeParams() {
+        return new ArrayList<>(types.get(getPrimaryType()));
```
```suggestion
        return types.get(getPrimaryType()).clone();
```
??

mcornaton committed:
Remove unnecessary calls to `toString`

mcornaton committed:
Remove unnecessary semicolons

mcornaton committed:
Remove unnecessary imports

mcornaton committed:
Make classes static to unload memory

mcornaton committed:
Remove unused lines

mcornaton committed:
Remove redundant variable initializations

mcornaton committed:
Replace `!Optional.isPresent` by `Optional.isEmpty`

mcornaton committed:
Remove redundant `new File` call

mcornaton committed:
Remove useless argument of `String.substring`

mcornaton committed:
Remove useless type arguments

mcornaton committed:
Replace `List.of(...).stream` by `Stream.of`

mcornaton committed:
Fix misc Bytes comments

mcornaton committed:
Fix `encodeByteArray` comparison

mcornaton committed:
Fix unnecessary imports

mcornaton committed:
Use diamond operator when possible

mcornaton committed:
Use method reference when possible and lambdas instead of Function creation

mcornaton committed:
Use Java11 methods

mcornaton committed:
Remove undocumented parameters in JavaDoc

mcornaton committed:
Remove unnecessary imports

mcornaton committed:
Clean up assertions

  - Use `assertFalse(...)` instead of `assertTrue(!...)`
  - Use `assertTrue(...)` instead of `assertEquals(true, ...)`

mcornaton committed:
Use diamond operator when possible

mcornaton committed:
Revert "Remove undocumented parameters in JavaDoc"

This reverts commit 07122e7a2171f023eb44242970b73376b352450d.

mcornaton committed:
Merge branch 'develop' into feature/update-deprecated-calls

# Conflicts:
#	src/test/java/com/iexec/common/task/TaskDescriptionTest.java

james-toussaint commented on src/main/java/com/iexec/common/chain/eip712/EIP712Entity.java:
```diff
@@ -43,7 +45,7 @@ protected EIP712Entity(EIP712Domain domain, M message) {
 
     @Override
     public HashMap<String, List<TypeParam>> getTypes() {
```
We can use map everywhere
(you can add a todo if you prefer)

mcornaton replied:
Thanks, I wondered whether I could use a `Map` there. That'll make things cleaner.
See https://github.com/iExecBlockchainComputing/iexec-common/pull/302/commits/490d6ad2e45e0a00aa4323007437ac990677a749.

james-toussaint commented on src/main/java/com/iexec/common/contract/generated/App.java:
```diff
@@ -20,6 +20,8 @@
 import java.util.Arrays;
```
You should not update what's inside /generated

james-toussaint replied:
If you have time you can setup sonar exclusion for such folder https://docs.sonarqube.org/latest/project-administration/narrowing-the-focus/

mcornaton replied:
Those are reverted.
That'd be great to set Sonar exclusion but we still can't sync our local Sonar settings with SonarQube so it would still annoy me :laughing:

mcornaton replied:
It should be OK in SonarQube:
![image](https://user-images.githubusercontent.com/87655102/149915832-6bc039d9-6569-4f29-9a45-e26f1d665d58.png)

james-toussaint commented on src/main/java/com/iexec/common/result/eip712/Eip712Challenge.java:
```diff
@@ -17,69 +17,49 @@
-@NoArgsConstructor
-public class Eip712Challenge {
-
+public class Eip712Challenge extends EIP712Entity<Message> {
```
:rocket:!

james-toussaint commented on src/main/java/com/iexec/common/utils/FileHelper.java:
```diff
@@ -420,11 +420,7 @@ private static void printFile(File file, int indent, StringBuilder sb) {
-            sb.append("│   ");
-        }
-        return sb.toString();
+        return "│   ".repeat(indent);
```
Nice

mcornaton replied:
Yeah, I didn't know it was a thing in Java 11 but that looks way better than the ugly loop :)

james-toussaint commented on src/test/java/com/iexec/common/sdk/order/payload/DatasetOrderTest.java:
```diff
@@ -25,10 +25,10 @@
 
     @Test
-    public void getDatasetOrder() throws JsonProcessingException {
+    void getDatasetOrder() throws JsonProcessingException {
```
Thanks for all!

mcornaton committed:
Use `Map` instead of `HashMap`

mcornaton committed:
Revert changes in /generated

mcornaton committed:
Revert changes of IexecHubContract

zguesmi commented on src/main/java/com/iexec/common/result/eip712/Eip712Challenge.java:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 IEXEC BLOCKCHAIN TECH
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
```
Let's keep it consistent with other files.

mcornaton committed:
Copyright 2020

mcornaton committed:
Merge branch 'develop' into feature/update-deprecated-calls
